homeassistant/components/fritz/common.py

```diff
@@ -981,6 +1011,50 @@ def device_info(self) -> DeviceInfo:
         )
 
 
+@dataclass
+class FritzRequireKeysMixin:
+    """Fritz entity description mix in."""
+
+    value_fn: Callable[[FritzStatus, Any], Any]
+
+
+@dataclass
+class FritzEntityDescription(EntityDescription, FritzRequireKeysMixin):
+    """Fritz entity base description."""
+
+
+class FritzBoxBaseCoordinatorEntity(update_coordinator.CoordinatorEntity):
+    """Fritz host coordinator entity base class."""
+
+    coordinator: AvmWrapper
+    entity_description: FritzEntityDescription
+
+    def __init__(self, avm_wrapper: AvmWrapper, device_name: str) -> None:
+        """Init device info class."""
+        super().__init__(avm_wrapper)
+        self.async_on_remove(
+            avm_wrapper.register_entity_updates(
+                self.entity_description.key, self.entity_description.value_fn
+            )
+        )
+        self._device_name = device_name
+        self._attr_name = f"{device_name} {self.entity_description.name}"
```

Comment:
Look at using `has_entity_name`

Reply:
I'm not sure what you mean 🤔

Reply:
Check in dev docs. Entity name is the way forward to define entity names with a combination of the entity name and device name

Reply:
reference [has_entity_name](https://developers.home-assistant.io/docs/core/entity/#has_entity_name-true-mandatory-for-new-integrations)

Reply:
Exactly 👍🏻

Reply:
nice one, didn't know about this before 👍

Reply:
Its hard to keep up with all additions :)